libs/MeshKernel/src/Polygons.cpp

```diff
@@ -145,11 +146,14 @@ std::vector<meshkernel::Point> Polygons::RefineFirstPolygon(size_t startIndex,
 
     if (endIndex <= startIndex)
     {
-        throw std::invalid_argument("Polygons::RefineFirstPolygon: The end index is smaller than the start index.");
+        throw std::invalid_argument("Polygons::RefineFirstPolygon: The end index is smaller than the start index: " + std::to_string(startIndex) + " >= " + std::to_string(endIndex));
```

Comment:
You can use out custom exceptions here, See Exceptions.hpp.

Reply:
I created a new exception ConstraintError and this will be raised for these constraint violations. It can (should) also be used in other constraint violation cases.

Reply:
Ok